hangmanlunch.rb

```diff
@@ -0,0 +1,208 @@
+require "colorize"
+
+class Hangman
+
+  def initialize
+    @answer_array = populate_answer
+    @guess_array = populate_guess_array
+    @paper = []
+    @top_row =      "|    _________"
+    @second_row =   "|     |/      "
+    @middle_row =   "|     |       "
+    @base_row =     "| ____|___ word: #{@guess_array.join(" ")}"
+    @misses = 0
+    @incorrect_array =[]
+
+    #remove this line to actually play-- leaving it for testing purposes
+    puts "Answer: " + @answer_array.join
+
+    start
+  end
+
+  # create paper
+  def draw_paper
+    @paper.each do |row|
+      puts row
+    end
+  end
+
+  def start
+    puts "\nLet's play Hangman!!\n"
+
+    #this kicks off the game loop in "play" method below
+    @now_playing = true
+
+    # creates empty scaffold (in the paper array)
+    @paper.push(@top_row, @second_row, @base_row)
+
+    5.times do
+      @paper.insert(2, @middle_row)
+    end
+
+    # draws initial empty scaffold
+    draw_paper
+
+    #starts asking for guesses
+    play
+  end
+
+  def play
+
+    while @now_playing
+      # asks for user guess and checks their guess
+      get_guess
+      check_guess(@letter_guess)
+    end
+  end
+
+
+  def get_guess
+    puts "\nGuess a letter! Or you can give up and type 'quit'."
+    # asks for user input (i.e. their letter guess)
+    @letter_guess = gets.chomp.downcase
+
+    acceptable_range = ("a".."z")
+
+    # allows user to quit the game
+    if @letter_guess == "quit"
+      exit
+
+    # controls for unexpected user input
+    elsif acceptable_range.include?(@letter_guess) == false
```

Comment:
This conditional is very important, but kind of awkward. It's hard to parse, mentally, that you should do something is this statement is false. Like, is this false? True. Ok. Wait, true so I shouldn't, or true that it's false so I should?

The if/elsif conditions aren't really related here, so if we break them into individual guard clauses, we can express the ideas more simply, like...

```ruby
exit if @letter_guess == "quit"
puts "We're working with letter here. Come on...".red.bold unless acceptable_range.include?(@letter_guess)
```

The `unless` in the second line lets us assert on something being false in what I hope is an easier to understand way. I used the if-at-the-end idiom because I'm lazy; these would work equally well in the normal `if/end` block structure.